csrc/host_ir/executor.cpp

```diff
@@ -20,7 +20,16 @@ HostIrExecutor::HostIrExecutor(
     HostIrExecutorParams params)
     : container_(std::move(container)),
       communicator_(communicator),
-      params_(params) {}
+      params_(params) {
+  auto device_index =
+      (communicator_ != nullptr && communicator_->is_available())
+      ? communicator_->deviceId()
+      : 0;
```

Comment:
This seems a wrong assumption. Even when `communicator_` not being available, the machine may still have multiple devices and the user may want to run the fusion on a particular device (and therefore not always device 0). 

@naoyam what's your suggestion? We could compute the device index from input tensors, similar to FusionExecutorCache. We could also make the device index a config of HostIrExecutor and therefore pass it into the constructor.

Reply:
right, let me know then.

Notice that the same situation is already present in the current code
https://github.com/NVIDIA/Fuser/blob/4e7365d18973819b9859f4b9ac498269257762a2/csrc/host_ir/executor.cpp#L49